mangaki/mangaki/utils/mal.py

```diff
@@ -120,17 +121,18 @@ def import_mal(mal_username, mangaki_username):
             score = int(entry.find('my_score').text)
             mal_id = entry.find('series_animedb_id').text
             try:
-                if Anime.objects.filter(title=title).count():
-                    anime = Anime.objects.get(title=title)
-                elif Anime.objects.filter(poster=poster).count() == 1:
-                    anime = Anime.objects.get(poster=poster)
-                elif Anime.objects.filter(poster=poster).count() >= 2:
-                    print('multiposters')
-                    raise Exception  # Report
-                else:
-                    entries = lookup_mal_api(title)
-                    retrieve_anime(entries)
-                    anime = Anime.objects.get(poster=poster)
+                try:
+                    anime = Work.objects.filter(category__slug='anime').get(title=title)
+                except Work.DoesNotExist:
+                    if Work.objects.filter(category__slug='anime', poster=poster).count() == 1:
+                        anime = Work.objects.get(poster=poster)
+                    elif Work.objects.filter(category__slug='anime', poster=poster).count() >= 2:
+                        print('multiposters')
+                        raise Exception  # Report
```

Comment:
`raise RuntimeException('Integrity violation: found two or more works with the same poster, do you come from the past?')`